### ISSUE TITLE
Add an option to the AWS plugin not to add fsx inline policy which is too permissive

### ISSUE THREAD

**Issue by descampsk** [kubeflow/kfctl#232]
When you use the AWS Plugin, kfctl will automaticaly add the [iam_csi_fsx_policy.json] policy to the worker nodes, even if we don't use the FSX CSI.

This policy is too permissive:

```
    {
      "Effect": "Allow",
      "Action": "s3:*",
      "Resource": "*"
    }, 
```

We need to delete manually this inline policy to be able to use Kubeflow in production for security reasons.

I would like to have an option (useFsx: false for example) to the AWS Plugin here: 
```
  - kind: KfAwsPlugin
    metadata:
      name: aws
    spec:
      useFsx: false
      auth:
        cognito:
          certArn: arn:aws:acm:us-west-2:xxxxx:certificate/xxxxxxxxxxxxx-xxxx
          cognitoAppClientId: xxxxxbxxxxxx
          cognitoUserPoolArn: arn:aws:cognito-idp:us-west-2:xxxxx:userpool/us-west-2_xxxxxx
          cognitoUserPoolDomain: your-user-pool
      region: us-west-2
      roles:
      - eksctl-kubeflow-aws-nodegroup-ng-a2-NodeInstanceRole-xxxxx
```

to allow kfctl to install or not the inline fsx policy.

In addition, it will be perfect if it's possible for all inlines policy.

**Comment by Jeffwan** [comment]
Thanks for the feedback. What about other components like EFS CSI plugin, etc? 
At the same time, we have pod IAM for service account feature. We plan to have one service account like `kf-admin` and assign all permission to components need to talk to AWS service. Is that accepatable?

**Comment by descampsk** [comment]
I check this PR https://github.com/kubeflow/kfctl/pull/225/files and I think it's a very cool feature but it won't fix my issue because this inline policy will be assign automatically to all roles provided in the plugins (If i correcly understand the changes)
For security reasons, we have to manage ourselves any iam policies including ingress alb policy or efs csi policy.
If i don't provide any roles in the AWS Plugin, does it work ? If yes, then I just have to give no roles and mange the needed policy by myself.
If not, then it will be cool to have a option managed_inline_policies: false to disable this comportment.

What do you think ?

**Comment by Jeffwan** [comment]
@descampsk 

If you use this feature, you don't have to specify roles in aws plugin. all components talking to AWS will use the service account with IAM policy. That means we won't add policies in node group role. You are protected.  Does this meet your need?  check here for more details. 
https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

**Comment by Jeffwan** [comment]
In latest kfctl, we don't apply fsx policies automatically. We will cut a new release for kfctl soon.

**Comment by descampsk** [comment]
Yeah I think it s fixed by 1.0.1.
Thanks I close.
